### PR TITLE
[TCFMM-40] Impose granularity in tick initialization

### PR DESCRIPTION
**Pull request by rinn7e**
[//]: # (This is a template of a pull request template.)
[//]: # (You should modify it considering specifics of a particular repository and put it there.)
[//]: # (Comments like this are meta-comments, they shouldn't be present in the final template.)
[//]: # (Comments starting with '

[//]: # (Here you can add a link to the corresponding issue tracker, e. g. https://issues.serokell.io/issue/AD-)
[//]: # (For GitHub/GitLab issues it is better to use just hash symbol or exclamation mark as it is more resistant to repo movements)
[//]: # (In this case please also prefix the link with "Resolves" keyword)
[//]: # (See https://docs.gitlab.com/ee/user/project/issues/managing_issues.html#closing-issues-automatically)
[//]: # (See https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords)
## Related issue(s)

Resolves TCFMM-40

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

[//]: # (Add more docs here if you have them in the repository)
- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [Specification](../tree/master/docs/specification.md)
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

[//]: # (Update link to style guide if necesary or remove if it's not present)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

**Comment by rinn7e** [comment]
@pasqu4le Just manage to work on this. This PR is ready for review. My only question is do we have to check the tick spacing on the tick_index_witness or not?

File: scripts/generate_error_code.hs [hunk 1]
```diff
@@ -87,6 +87,9 @@ invalidInputErrors = errorsEnumerate 100
   , "position_liquidity_below_zero_err"
       :? "Liquidity of a position went below zero."
 
+  , "incorrect_tick_spacing_err"
+      :? "Tick index is not multiple of tick spacing."
```
Comment (pasqu4le) [inline scripts/generate_error_code.hs:91]: Minor nitpick, I would change this to:
```suggestion
      :? "Tick indexes must be a multiple of the tick spacing."
```
to be in the same format the other errors.

File: Makefile [hunk 2]
```diff
@@ -96,6 +96,7 @@ $(OUT)/storage_%.tz: $(shell find ligo -name '*.mligo')
 			; y_token_id = $(y_token_id)n \
 			; x_token_address = (\"$(x_token_address)\" : address) \
 			; y_token_address = (\"$(y_token_address)\" : address) \
+			; tick_spacing = (1n : nat) \
```
Comment (pasqu4le) [inline Makefile:99]: This should be an argument of this target as well.

**Review by pasqu4le — COMMENTED** [review]
Overall LGTM, but the specification needs an update

**Comment by pasqu4le** [comment]
> do we have to check the tick spacing on the tick_index_witness or not?

I don't think so, witnesses can only be picked from initialized ticks (which are restricted) and the "infinite" values (which are to be accepted either way).

File: Makefile (hunk 2 above)
Comment (dcastro) [reply Makefile:99]: And it should be mentioned in `docs/compilation.md`

**Review by dcastro — APPROVED** [review]
Apart from the issues already raised by Pasquale, LGTM, so I'm approving it now.

File: scripts/generate_error_code.hs (hunk 1 above)
Comment (rinn7e) [reply scripts/generate_error_code.hs]: Updated this.

File: Makefile (hunk 2 above)
Comment (rinn7e) [reply Makefile]: Right, I meant to update this before pushing the PR but forgot. Also update the `compilation.md` as well.

**Review by pasqu4le — APPROVED** [review]
LGTM now :+1:

**Pull request merged**